src/Dialect/ONNX/ONNXOps/Math/ElementwiseBroadcast.cpp

```diff
@@ -161,6 +161,20 @@ LogicalResult ONNXBitShiftOp::inferShapes(
   return inferShapeForBroadcastingOps<ONNXBitShiftOp>(*this);
 }
 
+//===----------------------------------------------------------------------===//
```

Comment:
Can you motivate in a sentence why "castLike" is seen as a broadcast ? There is only one input.

Reply:
Ohh because Castlike actually has two inputs not one.

Reply:
I added a sentence :)

Reply:
Can you just check with a small example that if the two inputs are incompatible in shape, we don't have issues?
```
castlike( 32x64xf32, 2xi32)
```

that it works well with the broadcast op? I just don't remember if the broadcast logic is not taking over something that we don't really want (i.e. a "common" size between the various inputs). For example, an `add` operation with the above inputs is illegal as we cannot broadcast the one to the other.

Reply:
Okay I will put it back unary... I am fine with that.